src/main/java/org/dependencytrack/tasks/OsvDownloadTask.java

```diff
@@ -297,4 +308,26 @@ private boolean isVulnerabilitySourceClashingWithGithubOrNvd(String source) {
         return Vulnerability.Source.GITHUB.toString().equals(source)
                 || Vulnerability.Source.NVD.toString().equals(source);
     }
+
+    public static List<String> getEcosystems() {
+        ArrayList<String> ecosystems = new ArrayList<>();
+        String url = "https://osv-vulnerabilities.storage.googleapis.com/" + "ecosystems.txt";
+        request = new HttpGet(url);
+        try (final CloseableHttpResponse response = HttpClientPool.getClient().execute(request)) {
+            final StatusLine status = response.getStatusLine();
+            if (status.getStatusCode() == 200) {
+                try (InputStream in = response.getEntity().getContent();
+                     Scanner scanner = new Scanner(in, StandardCharsets.UTF_8.name())) {
+                    while (scanner.hasNextLine()) {
+                        ecosystems.add(scanner.nextLine().trim());
```

Comment:
Checking for blank lines may be a good idea here. Just so that we don't run into issues when the file ends with an empty line for example.